ui/components/app/snaps/snap-footer-button/snap-footer-button.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { FunctionComponent, MouseEvent as ReactMouseEvent } from 'react';
+import { ButtonVariant, UserInputEventType } from '@metamask/snaps-sdk';
+import { Button, ButtonSize, IconSize } from '../../../component-library';
+import {
+  AlignItems,
+  BorderColor,
+  Display,
+  FlexDirection,
+  IconColor,
+} from '../../../../helpers/constants/design-system';
+import { useSnapInterfaceContext } from '../../../../contexts/snaps';
+import { SnapIcon } from '../snap-icon';
+
+type SnapFooterButtonProps = {
+  name?: string;
+  isSnapAction?: boolean;
+  onCancel?: () => void;
+};
+
+export const SnapFooterButton: FunctionComponent<SnapFooterButtonProps> = ({
+  onCancel,
+  name,
+  children,
+  isSnapAction = false,
+  ...props
+}) => {
+  const { handleEvent, snapId } = useSnapInterfaceContext();
+
+  const handleSnapAction = (event: ReactMouseEvent<HTMLElement>) => {
+    event.preventDefault();
+
+    handleEvent({
+      event: UserInputEventType.ButtonClickEvent,
+      name,
+    });
+  };
+
+  const handleClick = isSnapAction ? handleSnapAction : onCancel;
+
+  return (
+    <Button
```

Comment:
Colors, hover state etc to be confirmed by @eriknson

Reply:
Message sent